### PR TITLE
Allow to use external io_service in restbed::Service

### PR DESCRIPTION
**Pull request by chronoxor**

### CODE REVIEW

**Comment by ben-crowhurst** [comment]
Shall review and comment, thanks.

File: source/corvusoft/restbed/service.cpp [hunk 1]
```diff
@@ -176,29 +176,32 @@ namespace restbed
...
         {
-            m_pimpl->m_io_service->post( m_pimpl->m_ready_handler );
+            m_pimpl->get_io_service()->post( m_pimpl->m_ready_handler );
         }
         
         m_pimpl->m_uptime = steady_clock::now( );
         unsigned int limit = m_pimpl->m_settings->get_worker_limit( );
         
-        if ( limit > 0 )
+        if ( m_pimpl->m_external_io_service == nullptr )
```
Comment (chronoxor) [inline source/corvusoft/restbed/service.cpp]: Currently I host WebServer in separate thread with separate io_service. But I believe this solution will scale for multiple threads by hosting each WebServer with io_service in a new thread using reuse_address(true).

Comment (chronoxor) [inline source/corvusoft/restbed/service.cpp]: This allows me to host different Asio based servers in a single io_service - TCP, UDP, WebSockets, HTTP/HTTPS. If you are interesting you may look into my CppServer project - https://github.com/chronoxor/CppServer/tree/master/source/server/asio

**Comment by ben-crowhurst** [comment]
Is this feature still of interest? We are looking to merge this week.

**Comment by chronoxor** [comment]
Yes! I use this feature in production, but currently with my local fork. I'll move back to your repo again after merge!

**Comment by silviucpp** [comment]
Any news on this feature?

**Comment by ben-crowhurst** [comment]
If anyone is will to resolve the outstanding branch conflicts we will this change into master.

**Comment by chronoxor** [comment]
Updated pull request was sent

**Pull request closed without merging**